pkg/kubelet/qos/memory_policy.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package qos
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+)
+
+const (
+	PodInfraOomAdj int = -999
+)
+
+// isMemoryBestEffort returns true if the container's memory requirements are best-effort.
+func isMemoryBestEffort(container *api.Container) bool {
+	// A container is memory best-effort if its memory request is unspecified or 0.
+	// If a request is specified, then the user expects some kind of resource guarantee.
+	return container.Resources.Requests.Memory().Value() == 0
+}
+
+// isMemoryGuaranteed returns true if the container's memory requirements are Guaranteed.
+func isMemoryGuaranteed(container *api.Container) bool {
+	// A container is memory guaranteed if its memory request == memory limit.
+	// If memory request == memory limit, the user is very confident of resource consumption.
+	return container.Resources.Requests.Memory().Value() == container.Resources.Limits.Memory().Value()
+}
+
+// GetContainerOomAdjust returns the amount by which the OOM score of all processes in the
+// container should be adjusted. The OOM score of a process is the percentage of memory it consumes
+// multiplied by 100 (barring exceptional cases) + a configurable quantity which is between -1000
+// and 1000. Containers with higher OOM scores are killed if the system runs out of memory.
+func GetContainerOomScoreAdjust(container *api.Container, memoryCapacity int64) int {
```

Comment:
I added unit tests for isMemoryGuaranteed and isMemoryBestEffort (which essentially tests the first 2 cases below). I didn't add tests for Burstable containers because the precise formula might be in flux, but I can add them if you think I should!